### PR TITLE
Task 645 - 648: Wiki auto and winograd dataset

### PR DESCRIPTION
**Pull request by khushal1996**

### CODE REVIEW

**Comment by danyaljj** [comment]
@yeganehkordi wondering if you can review this PR?

**Comment by swarooprm** [comment]
@SavanDoshi's reviews: https://github.com/swarooprm/CSE576HW/pull/77

**Comment by swarooprm** [comment]
@khushal1996, your tasks look good. But there are certain changes you need to be done before your tasks get merged:
- task 645- I would call it a "topic" rather than a summary. Also the definition should be written like how you will instruct your friend to do the task.
Here's the suggested definition: "Generate a topic word for the given sentence. A topic word is usually a name of a person, place, animal or thing the sentence is focusing on"
- task 646- suggested definition: "Generate a pronoun present in the given sentence. Some examples of pronouns are they, them, you, he, his, her, you". The second positive example does not look good (specially the phrase "because he was it"), rephrase. Also, I guess "he" should be the correct pronoun here: check
- task 647-suggested definition "Generate a quote where the pronoun is used in the given statement. Quote must be a part of the given statement." Typo: madonnas boyfriend-->madonna's boyfriend
- task 648-suggested definition "generate the word in the sentence to which the pronoun in the input is referring". Positive and negative examples are incorrect here (they are same as last task), replace them with new examples. Instances are also incorrect and need to be something like this:
{
            "input": "text: the city councilmen refused the demonstrators a permit because they feared violence.; pronoun: they",
            "output": [
                "the city councilmen"
            ]
        },
        {
            "input": "text: the city councilmen refused the demonstrators a permit because they advocated violence.; pronoun: they",
            "output": [
                "the demonstrators"
            ]
        }

You should be able to fix these quickly as I have written the suggested definition which you need to directly paste. Let me know once you are done and I will verify.

**Comment by danyaljj** [comment]
@khushal1996 wondering if you can address Swaroop's suggestions above?

**Comment by danyaljj** [comment]
Please make sure to set the `"Categories": [""]` and `"Domains": [""]` fields according to [the taxonomy presented here](https://github.com/allenai/natural-instructions-expansion/blob/master/doc/task-hierarchy.md).

File: tasks/task645_summarization.json [hunk 1]
```diff
@@ -0,0 +1,12039 @@
+{
+    "Contributors": [
+        "khushal chandresh modi"
+    ],
+    "Source": [
+        "wiki-auto-all-data",
+        "https://www.dropbox.com/sh/ohqaw41v48c7e5p/aadb6_qwcwghvscqg121ck9ma?dl=0"
+    ],
+    "Categories": [
+        "topic"
```
Comment (danyaljj) [inline tasks/task645_summarization.json]: `Topic Generation`

File: tasks/task646_answer_generation.json [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+            ]
+        },
+        {
+            "input": "the lawyer asked the witness a question, but he was reluctant to repeat it.",
+            "output": [
+                "he"
+            ]
+        },
+        {
+            "input": "the lawyer asked the witness a question, but he was reluctant to answer it.",
```
Comment (danyaljj) [inline tasks/task646_answer_generation.json:97]: There are many examples where there is more than one pronoun in the sentence. 
Like here, the gold output should be `["it", "he"]`.

**Comment by swarooprm** [comment]
@khushal1996, can you fix the issue Danie pointed out in instances?

**Comment by swarooprm** [comment]
> @khushal1996, can you fix the issue Danie pointed out in instances?

a gentle reminder! @khushal1996

**Comment by danyaljj** [comment]
Applied minor improvements to the task descriptions. 
Moved one suggestion to #488. 
Merging when things turn green.

**Pull request merged**